rustls/src/lib.rs

```diff
@@ -35,14 +35,6 @@
 //!
 //! ### targets with no atomic ptr
 //!
-//! This fork supports using `Arc` from `portable-atomic-util` to support targets with no atomic ptr, with the following requirements:
-//!
-//! Must use Rust nightly toolchain.
-//!
-//! Must use the following cfg flags in `RUSTFLAGS` FOR `cargo build` (etc.):
-//! - `--cfg portable_atomic_unstable_coerce_unsized`
```

Comment:
I think most of this info still belongs here, it was quoted near the beginning of README.md and I think most of this is still true. Also not just for targets with no atomic ptr, now true by default with these changes. Also some info below seems to be important for no-std in general, I think an update is needed to document this.